src/rez/build_system.py

```diff
@@ -104,12 +127,13 @@ class BuildSystem(object):
     """A build system, such as cmake, make, Scons etc.
     """
     @classmethod
-    def name(cls):
+    def name(cls) -> str:
         """Return the name of the build system, eg 'make'."""
         raise NotImplementedError
 
-    def __init__(self, working_dir, opts=None, package=None,
-                 write_build_scripts=False, verbose=False, build_args=[],
+    def __init__(self, working_dir: str, opts=None,
```

Comment:
I'd prefer for new annotations to be done in a new PR, so that we can keep the scope of this PR under control.  There are literally thousands of unannotated arguments that still need to be dealt with.

Reply:
would like to avoid scope creep.